### DISCUSSION TITLE
Specify desired material on .glb Import (at runtime)

### DISCUSSION THREAD

**Discussion by BertrandDeSaintBezier** [atteneder/glTFast#674] [Q&A]
As the title says, this is maybe super obvious but I'm just missing it : how can I specify which material should be used when importing a .glb file at runtime ? Is there an easy way of doing this ?

**Comment by atteneder** [comment]
The way glTFast works is it pre-selects the required shader based on the glTF material's features and creates a new material.

Users can inject their own logic by providing an implementation of [`IMaterialGenerator`](https://docs.unity3d.com/Packages/com.unity.cloud.gltfast@6.7/api/GLTFast.Materials.IMaterialGenerator.html) to [`GltfImport`'s constructor](https://docs.unity3d.com/Packages/com.unity.cloud.gltfast@6.7/api/GLTFast.GltfImport.html#constructors).

hth
